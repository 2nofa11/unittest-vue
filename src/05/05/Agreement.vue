<script setup lang="ts">
const checked = defineModel<boolean>();
</script>

<template>
  <fieldset data-test="agreement-fieldset">
    <legend data-test="agreement-legend">利用規約の同意</legend>
    <label>
      <input type="checkbox" v-model="checked" data-test="agreement-checkbox" />
      当サービスの<a href="/terms">利用規約</a>を確認し、これに同意します
    </label>
  </fieldset>
</template>
