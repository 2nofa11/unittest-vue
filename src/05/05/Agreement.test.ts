import { describe, expect, test } from "@jest/globals";
import { mount } from "@vue/test-utils";
import Agreement from "./Agreement.vue";

describe("Agreement.vue", () => {
  test("fieldset が表示され、legend が含まれる", () => {
    const wrapper = mount(Agreement, { props: { modelValue: false } });
    const fieldset = wrapper.find('[data-test="agreement-fieldset"]');
    expect(fieldset.exists()).toBe(true);
    expect(fieldset.find("legend").text()).toBe("利用規約の同意");
  });

  test("チェックボックスはチェックが入っていない", () => {
    const wrapper = mount(Agreement, { props: { modelValue: false } });
    const checkbox = wrapper.find<HTMLInputElement>(
      '[data-test="agreement-checkbox"]'
    );
    expect(checkbox.exists()).toBe(true);
    expect(checkbox.element.checked).toBe(false);
  });

  test("利用規約へのリンクがある", () => {
    const wrapper = mount(Agreement, { props: { modelValue: false } });
    const link = wrapper.find("a");
    expect(link.exists()).toBe(true);
    expect(link.text()).toBe("利用規約");
    expect(link.attributes("href")).toBe("/terms");
  });

  test("Snapshot: 利用規約の同意が表示される", () => {
    const wrapper = mount(Agreement, { props: { modelValue: false } });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
