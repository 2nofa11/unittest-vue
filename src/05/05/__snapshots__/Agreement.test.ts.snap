// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Agreement.vue Snapshot: 利用規約の同意が表示される 1`] = `
"<fieldset data-test="agreement-fieldset">
  <legend data-test="agreement-legend">利用規約の同意</legend><label><input type="checkbox" data-test="agreement-checkbox"> 当サービスの<a href="/terms">利用規約</a>を確認し、これに同意します </label>
</fieldset>"
`;
